Cherry-pick node creation usability enhancements (2.1)

RandomShaper opened the pull request:
Namely, favorites, recents and class description at the node creation dialog.

reduz committed:
Making bits of docs appear in different dialogues is made easier with EditorHelpBit

RandomShaper committed:
Add EditorNode::set_visible_editor()
Cherry-picked from d8af6330e57b98f370ed56d891a6a6952957cfec

akien-mga commented on tools/editor/create_dialog.cpp:
```diff
@@ -42,11 +42,81 @@
+	} else {
+		if (base_type=="Node") {
+			//harcode some favorites :D
+			favorite_list.push_back("Panel");
```
That part was revert in master, so it should not be added to 2.1 IMO.

RandomShaper replied:
You're right. I've pushed with the sub-cherry-pick added.

RandomShaper committed:
Added favorites and recent history to create dialog
Cherry-picked from 8d785812351a0a3eca88214f2436bb7574d55873 and 827a9aa8294e7e2405f645579cc3e7044f3be079
